src/test/java/com/checkmarx/flow/cucumber/integration/end2end/genericendtoend/BugTracker.java

```diff
@@ -110,11 +109,101 @@ void deleteIssues() {
                     deleteIssues();
                 }
             } catch (Exception e) {
-                log.warn("Failed to clean tickets from Jira");
+                log.error("Failed to clean tickets from Jira {}", e.getMessage());
             }
         }
+    },
+    GitLab{
+        private static final String GET_PROJECT_URL = "/projects?search={name}";
+        private static final String GET_ISSUES_URL = "/projects/{id}/issues/";
+        private static final String DELETE_ISSUE_URL = "/projects/{id}/issues/{issueId}";
+
+        private final RestTemplate restTemplate = new RestTemplate();
+        private String projectName = null;
+        private  Integer projectId = null;
+        private  GitLabProperties gitLabProperties;
+        @Override
+        void init(GenericEndToEndSteps genericEndToEndSteps) {
+            gitLabProperties = genericEndToEndSteps.gitLabProperties;
+            projectName = genericEndToEndSteps.getRepository().getRepoName();
+            projectId = getProjectId();
+        }
+
+        @Override
+        void verifyIssueCreated(String severities, String engine) {
+            for (int retries = 0; retries < NUMBER_OF_RETRIES; retries++) {
+                try {
+                    TimeUnit.SECONDS.sleep(RETRY_TIMEOUT_IN_SECONDS);
+                    log.info("checking for issues in gitlab project {} starting attempt {}", projectId, retries + 1);
+                } catch (Exception e) {
+                    log.error("error in timeout while verifying created issues: {}", e.getMessage());
+                }
+                JSONArray issuesList = getAllGitlabIssues();
+                if (issuesList.length() > 0){
+                    log.info("successfully verified created issues!");
+                    return;
+                }
+            }
+            String msg = String.format("didn't find any security issues in gitlab project %d after %d retries", projectId, NUMBER_OF_RETRIES);
+            log.error(msg);
+            fail(msg);
```

Comment:
It's better to have something like assertTrue() here instead of fail() to make the test more readable. This way it will be clear what exactly we are checking.

Reply:
changed to assertTrue()